qa/rpc-tests/test_framework/mininode.py

```diff
@@ -359,57 +359,131 @@ def __repr__(self):
             % (repr(self.prevout), bytes_to_hex_str(self.scriptSig),
                self.nSequence)
 
-def setCTxOutValue(amount):
-    vchCommitment = [0]*33
-    for i in range(8): #8 bytes
-        vchCommitment[33-1-i] = ((amount >> (i*8)) & 0xff)
-    return ''.join(map(chr, vchCommitment))
+class CTxOutAsset(object):
+    def __init__(self, vchCommitment=b"\x00", vchRangeProof=b"", vchNonceCommitment=b""):
+        self.vchCommitment = vchCommitment
+        self.vchRangeProof = vchRangeProof
+        self.vchNonceCommitment = vchNonceCommitment
+
+    def deserialize(self, f):
+        version = ord(f.read(1))
+        if version == 0: self.vchCommitment = b'\x00'
+        elif version == 1: self.vchCommitment = b'\x01' + f.read(32)
+        elif version == 0xff: self.vchCommitment = b'\xff' + f.read(32)
+        elif version == 10 or version == 11: self.vchCommitment = bytes([version]) + f.read(32)
+        else: raise 'invalid CTxOutAsset in deserialize'
+
+    def serialize(self):
+        r = b""
+        r += self.vchCommitment
+        return r
+
+    def setToAsset(self, val):
+        if len(val) != 32:
+            raise 'invalid asset hash (expected 32 bytes got %d)' % len(val)
+        self.vchCommitment = b'\x01' + val
+
+    def __repr__(self):
+        return "CTxOutAsset(vchCommitment=%s vchRangeProof=%s, vchnonceCommitment=%s)" \
+            % (self.vchCommitment, self.vchRangeProof,
+               self.vchNonceCommitment)
 
-#Serialization only does value commitment
 class CTxOutValue(object):
-    def __init__(self, vchCommitment=[], vchRangeProof="", vchNonceCommitment=""):
+
+    def __init__(self, value=None):
+        self.setNull()
+        if value != None: self.setToAmount(value)
+
+    def setNull(self):
+        self.vchCommitment = b'\x00'
+        self.vchRangeProof =b''
```

Comment:
rangeproof no longer resides in the output. It resides in the `CTxWitness` based field `wit` in `CTransaction`